### PR TITLE
Add semantic functions for accessing components of Cartesian products

### PR DESCRIPTION
**Pull request by zhivkag**
This is a step towards unifying usage of pattern matching vs getters for Cartesian products. Whatever we chose to define for this will be used to update the corresponding rules.

### CODE REVIEW

**Comment by zhivkag** [comment]
I am wondering if we need to add lookupSetter/lookupGetter, since when we need to lookup a field, it need to return the corresponding implicit getter or setter, depending if it is a property extraction or assignment.
According to Dart doc the implicit setters names have a '=' appended to it, but not in DartKernel. What do you think?

**Comment by chloestefantsova** [comment]
I think having lookupSetter/lookupGetter is a good idea.  Not sure about the suffix '=' though.  Do you have a specific example in mind where it may be important for the semantics?

File: documentation/kernel-semantics.tex [hunk 1]
```diff
@@ -129,6 +129,12 @@
 \newcommand{\dsemstring}{\mathbf{String}}
 \newcommand{\dsemsymbol}{\mathbf{Symbol}}
 
+% Functions for projection on components of class.
+\newcommand{\superclass}[1]{\textit{superclass}({#1})}
+\newcommand{\members}[1]{\textit{members}(#1)}
+\newcommand{\lookupMember}[2]{\textit{lookupMember}(#1,\,#2)}
+\newcommand{\class}[1]{\textit{class}(#1)}
```
Comment (chloestefantsova) [inline documentation/kernel-semantics.tex]: I think these should be defined with \DeclareMathOperator.

**Review by chloestefantsova — APPROVED** [review]
LGTM with one small comment.

**Pull request merged**